libs/modkit-utils/src/humantime_serde.rs

```diff
@@ -0,0 +1,242 @@
+#![forbid(unsafe_code)]
+
+//! Serde support for the `humantime` crate.
+//!
+//! Based on [this fork](https://github.com/jean-airoldie/humantime-serde).
+//!
+//! Currently `std::time::Duration` is supported.
+//!
+//! # Example
+//! ```
+//! use serde::{Serialize, Deserialize};
+//! use std::time::Duration;
+//!
+//! #[derive(Serialize, Deserialize)]
+//! struct Foo {
+//!     #[serde(with = "modkit_utils::humantime_serde")]
+//!     timeout: Duration,
+//! }
+//! ```
+
+use std::fmt;
+use std::time::Duration;
+
+use humantime;
+use serde::{de, Deserializer, Serializer};
+
+/// Deserializes a `Duration` via the humantime crate.
+///
+/// This function can be used with `serde_derive`'s `with` and
+/// `deserialize_with` annotations.
+/// # Errors
+/// Returns a `humantime::Error` if string is not a valid Duration
```

Comment:
I think comment doesn't reflect reality. Looks like a formatting cannot fail only the serializer can return an error